gallery/visualtests/nxMultiFileUpload.js

```diff
@@ -0,0 +1,545 @@
+
+      const [selectedFile3] = (await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(3)`)),
+          textContent3 = await selectedFile3.evaluate(e => e.textContent);
+
+      const [selectedFile4] = (await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(4)`)),
+          textContent4 = await selectedFile4.evaluate(e => e.textContent);
+
+      expect(textContent1).toMatch('14.0 B');
+      expect(textContent2).toMatch('2.0 kB');
+      expect(textContent3).toMatch('1.5 MB');
+      expect(textContent4).toMatch('1.2 GB');
+    });
+
+    it('removes the selected file when the dismiss button is clicked', async function() {
+      await openFileChooser([files.bytes, files.kilobytes]);
+
+      let [selectedFile1, dismissBtn1] = await waitAndGetElements(
+              `${complexExampleSelector} .nx-selected-file:nth-child(1)`,
+              `${complexExampleSelector} .nx-selected-file:nth-child(1) .nx-selected-file__dismiss-btn`
+          ),
+          textContent1 = await selectedFile1.evaluate(e => e.textContent);
+
+      const [selectedFile2] = await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(2)`),
+          textContent2 = await selectedFile2.evaluate(e => e.textContent);
+
+      // confirm layout of selected files
+      expect(textContent1).toMatch('14.0 B');
+      expect(textContent2).toMatch('2.0 kB');
+
+      await dismissBtn1.click();
+
+      [selectedFile1] = await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(1)`);
+      textContent1 = await selectedFile1.evaluate(e => e.textContent);
+
+      // confirm the correct file was removed
+      expect(await isInDocument(selectedFile2)).toBe(false);
+      expect(await isInDocument(selectedFile1)).toBe(true);
+      expect(textContent1).toMatch('2.0 kB');
+    });
+
+    it('sets the no File Selected message when the last selected file is removed', async function() {
+      await openFileChooser([files.bytes]);
+
+      const [selectedFile, dismissBtn] = await waitAndGetElements(
+          `${complexExampleSelector} .nx-selected-file`,
+          `${complexExampleSelector} .nx-selected-file .nx-selected-file__dismiss-btn`
+      );
+
+      await dismissBtn.click();
+
+      const [noFileSelectedMessage] = await waitAndGetElements('.nx-multi-file-upload__no-file-message'),
+          textContent = await noFileSelectedMessage.evaluate(e => e.textContent);
+
+      expect(await isInDocument(selectedFile)).toBe(false);
+      expect(await isInDocument(noFileSelectedMessage)).toBe(true);
+      expect(textContent).toBe('No file selected');
+    });
+
+    describe('focus behavior', function() {
+      it('sets focus on the appropriate dismiss button once a file is removed', async function() {
+        await openFileChooser([files.bytes, files.kilobytes, files.megabytes, files.gigabytes]);
+
+        let [selectedFile1, dismissBtn1] = await waitAndGetElements(
```

Comment:
Each `waitAndGetElements` call is a separate round-trip communication from the test process to the browser.  It makes a noticeable difference to the amount of time the tests take if they are grouped together as much as possible.  These three should definitely be grouped together into a single `waitAndGetElements`.  There are some in other tests that could probably be grouped better as well

Reply:
> Each `waitAndGetElements` call is a separate round-trip communication from the test process to the browser. It makes a noticeable difference to the amount of time the tests take if they are grouped together as much as possible. These three should definitely be grouped together into a single `waitAndGetElements`. There are some in other tests that could probably be grouped better as well

Combined waitAndGetElements here: [1ef9e36](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/1ef9e36f7cf4a15b52a572334558e2cc9d3b0d89)